blockchain/blockchain-wallet-v4-frontend
# feat(bitpay): implement BitPay's json payment protocol

## Description (optional)
Support opening BitPay invoices in the web wallet through bitcoin linking. 

## Testing Steps (optional)
1. Login to bitpay.com (slack me I can give you a login)
2. Go to https://bitpay.com/dashboard/payment-tools/payment-buttons
3. Scroll to the bottom and click the Pay with BitPay button to generate a new BitPay invoice.
4. When the invoice opens, select Bitcoin, then click the QR code that pops up to copy the bitcoin:?r=... link. (Their pay in wallet button functionality is broken, have to manually copy bitcoin link)
5. Paste this bitcoin:?r=... link into a new tab, wait for redirect, login when prompted, and wait for send modal popup. 
6. (you must also have Bitcoin link handling enabled in settings)

Testing details:
All inputs are disabled except for the description detailing the invoice. This lets users optionally include more details to specify what sort of thing they are buying from the merchant. 
(I've enabled the amount field for now as there is a bug in production @schnogz said he would check out where the amount and description are not being passed to the second step confirm screen)
A countdown timer is included above the To field to give the user feedback on how long they have to pay the invoice. 

Tests:
1. In order to bypass the amount bug, delete and retype the exact Bitcoin amount in the amount field, click continue to go to the second step confirm screen and complete the payment, BitPay invoice will reflect payment promptly.
2. Wait for the invoice to expire which will cause the send modal to close and a BitPay invoice expired info modal to popup informing the user that the invoice has expired.
